src/main/java/me/ryanhamshire/GriefPrevention/EntityEventHandler.java

```diff
@@ -988,6 +968,22 @@ else if(attackerData.pvpImmune)
                                 if(sendErrorMessagesToPlayers) GriefPrevention.sendMessage(attacker, TextMode.Err, Messages.CantFightWhileImmune);
                                 return;
                             }
+                            // disallow players attacking tamed wolves (dogs) unless under attack by said wolf
+                            else if (tameable instanceof Wolf) {
+                                if (!tameable.getOwner().equals(attacker)) {
```

Comment:
Does `Tameable#getOwner` use the current `Player` object when the owner is online? Big_Scary compares UUID earlier in this method.

Reply:
Oh shoot, that is right, didn't think about the offline player. Ill make that work

Reply:
Oh wait I just re-read it. It wouldn't matter, its just checking to make sure that the player attacking the wolf is not the owner.

Reply:
`attacker` is a `Player` whereas the instance returned by getOwner is `AnimalTamer`, which could be implemented by OfflinePlayer or Player (and another class that I don't recall). My hunch is that it returns an OfflinePlayer instance similar to Bukkit#getOfflinePlayer, which will _not_ be the same instance as the current `Player` instance.

You can test this by taming an animal, logging off (or better yet, restart server), then log on and see if getOwner matches Player

Reply:
Ive read over your message a few times now (ok a lot of times) and Im confused. I get the logic behind what you're saying, but I'm confused at the issue.
This condition is just to check, when a player is hitting a dog, if they are not the owner, it will prevent them from hurting the dog, unless the dog is attacking them.

I have tested this both with the owner online and offline. Both after restarts and theres been no issues. Everything works swimmingly. 

If I am totally missing the point here please let me know, because Im very confused. If you have a suggestion on a change that would be great too :)